Client/Frontend/Browser/TabManager.swift

```diff
@@ -453,14 +453,38 @@ class TabManager: NSObject {
                     for delegate in self.delegates {
                         delegate.get()?.tabManagerDidAddTabs(self)
                     }
+                } else {
+                    self.showFocusPromoToast()
                 }
                 self.eraseUndoCache()
             })
         }
 
         delegates.forEach { $0.get()?.tabManagerDidRemoveAllTabs(self, toast: toast) }
     }
-    
+
+    func showFocusPromoToast() {
+        // add check for event stuff
+        guard let shouldShowFocusUI = LeanplumIntegration.sharedInstance.getBoolVariableFromServer(key: "shouldShowFocusUI"), let focus = NSURL(string: "firefox-focus://"), let klar = NSURL(string: "firefox-klar://"), !UIApplication.shared.canOpenURL(focus as URL) && !UIApplication.shared.canOpenURL(klar as URL) && shouldShowFocusUI else {
+            return
+        }
+
+
+        var toast: LeanplumPromoButtonToast
+        let templateDict = LeanplumIntegration.sharedInstance.getTemplateDictionary()!
+        toast = LeanplumPromoButtonToast(labelText: templateDict["Template Text"]! , buttonText: templateDict["Button Text"]!, colorText: templateDict["Hex Color String"]!, completion: { buttonPressed in
+            guard let url = NSURL(string: "https://itunes.apple.com/app/id1055677337") else {
```

Comment:
Why are you confining the URL here to this URL? It should be set in the template dictionary?